hibernate-core/src/main/java/org/hibernate/type/TypeHelper.java

```diff
@@ -159,6 +159,9 @@ else if ( row[i] == LazyPropertyInitializer.UNFETCHED_PROPERTY || row[i] == Prop
 			if ( original[i] == LazyPropertyInitializer.UNFETCHED_PROPERTY || original[i] == PropertyAccessStrategyBackRefImpl.UNKNOWN ) {
 				copied[i] = target[i];
 			}
+			else if ( target[i] == LazyPropertyInitializer.UNFETCHED_PROPERTY ) {
+				copied[i] = types[i].replace( original[i], null, session, owner, copyCache );
```

Comment:
It a little ugly to knowingly do the wrong thing here if `LEGACY_LOB_MERGE_STRATEGY` is used. I think it would be reasonable to throw `UnsupportedOperationException` here if `LEGACY_LOB_MERGE_STRATEGY` is used. Unfortunately, it's not possible because `Dialect#LEGACY_LOB_MERGE_STRATEGY` is protected.